### PR TITLE
Fix test use date dynamically

### PR DESCRIPTION
**Pull request by lomamech**
- Use of dynamic dates
- Skip csv file import tests
- Modification of the values of the test results

### CODE REVIEW

**Comment by jmcameron** [comment]
How did you test this to verify that it works for a variety of dates?

**Comment by jmcameron** [comment]
Great job, @lomamech !!

I will take a look at it, but I suggest you ask @jniles to review and release it so we can get some of these other PRs released soon!

File: server/controllers/stock/core.js [hunk 1]
```diff
@@ -755,9 +754,8 @@ async function getInventoryQuantityAndConsumption(params, monthAverageConsumptio
     let clone = moment(startDate);
     const months = (nbrMonth - 1 > -1 && nbrMonth) ? nbrMonth - 1 : 0;
     clone = clone.subtract(months, 'month').toDate();
-    const m = clone.getMonth() + 1;
-    const y = clone.getFullYear();
-    startDate = new Date(`${y}-${m}-01`);
+
+    startDate = new Date(clone);
```
Comment (jniles) [inline server/controllers/stock/core.js:758]: Excellent  :+1:

**Review by jmcameron — COMMENTED** [review]
FYI, I had this failure on my pc (MySQL 8, latest node, Bhima_test):
```
  1) (/purchases) Purchases
       GET /inventory/metadata/:uuid downloads
    Checking the calculation of the order interval for inventory: Prednisolone, 0,5%, Solution, Flacon, Unité:

      AssertionError: expected 3.78 to equal 3.77
      + expected - actual

      -3.78
      +3.77

      at /home/jmcameron/Bhima/test/integration/purchase.js:253:50
      at processTicksAndRejections (internal/process/task_queues.js:97:5)
```

This was odd -- off by one cent.  Most errors are larger.

Also, the End-to-End tests produced an error:
```
  1) Update Employees
       should update data for employee:
     NoSuchElementError: No element found using locator: By(css selector, [data-row="EM.TE.1"])
```

I'm not sure how to treat these errors on my local system.  Since they passed via Travis CI, I'm not sure how important my errors are.

File: test/end-to-end/purchases/purchases.spec.js [hunk 2]
```diff
@@ -19,11 +20,12 @@ describe('Purchase Orders', () => {
 
   it('supports single item purchase orders', async () => {
     const page = new PurchaseOrderPage();
+    const datePurchase = moment(new Date(), 'YYYY-MM-DD').subtract(1712, 'days');
```
Comment (jniles) [inline test/end-to-end/purchases/purchases.spec.js]: By the way, it's helpful to make moment into a date by calling `.toDate()` on the moment object.  Like this:
```js
const datePurchase = moment(new Date(), 'YYYY-MM-DD').subtract(1712, 'days').toDate();
```

File: test/end-to-end/stock/stock.spec.js [hunk 3]
```diff
@@ -17,7 +17,7 @@ describe('Stock end-to-end test', () => {
   describe('Stock Assign Module', StockAssignTests);
   describe('Stock Entry Test', StockEntryTests);
   describe('Stock Exit Test', StockExiTests);
-  describe('Stock Import', StockImportTests);
+  describe.skip('Stock Import', StockImportTests);
```
Comment (jniles) [inline test/end-to-end/stock/stock.spec.js:20]: TODO :)

File: test/end-to-end/stock/stock.z1.inventory-adjustment.js [hunk 4]
```diff
@@ -25,16 +25,16 @@ function StockInventoryAdjustmentTests() {
...
     // set all other to zero
-    for (let i = 0; i < 16; i++) {
+    for (let i = 0; i < 5; i++) {
       // eslint-disable-next-line
       await page.setQuantity(i, 5, 0);
     }
 
-    // set the VITAMINE-A to 23
-    await page.setQuantity(3, 5, 23);
+    // set the Vitamines B1+B6+B12 to 23
```
Comment (jniles) [inline test/end-to-end/stock/stock.z1.inventory-adjustment.js:33]: Nice 👍

File: test/integration/purchase.js [hunk 5]
```diff
@@ -9,11 +9,29 @@ const SearchTests = require('./purchase.search.js');
  *
  * This test suite implements full CRUD on the /purchases HTTP API endpoint.
  */
-describe('(/purchases) Purchases', () => {
+describe.only('(/purchases) Purchases', () => {
```
Comment (jniles) [inline test/integration/purchase.js]: We need to remove this ".only()" call.

**Review by jniles — COMMENTED** [review]
Changes look good.  Can you remove the `.only()` call so we can be sure the tests pass?

**Comment by jniles** [comment]
Excellent.  Looks good to me!

bors r+

**Comment by jniles** [comment]
Awesome.  Thanks so much @lomamech !

**Pull request merged**